### PR TITLE
Publish AMI to multiple regions

### PR DESCRIPTION
**Pull request by ghost**
I added a couple params to the pipeline:

1. `CONJUR_VERSION`, which defaults to 'latest'. This means a new AMI will be created every day, from conjur latest.
2. `PROMOTE_TO_REGIONS`, default to `false`. If a build is run with this flipped on, the AMI will be shared across regions.

So the normal workflow is that AMIs are created from conjur latest, which is stable. When we want to do a release, we can trigger this pipeline with `CONJUR_VERSION` = release version and `PROMOTE_TO_REGIONS` to true.

We need to tackle AMI management with this change. Even if we only build AMIs every week we will need a reaper to go through and clear out the old ones. Otherwise this will become quickly unmanageable. Perhaps we can use https://github.com/bonclay7/aws-amicleaner? I think this should be a separate issue.

Also, we need to figure out how to share the AMIs with our users. Ideally we'd have an always up to date list on conjur.org. The mechanics of that are what we need to figure out.

### CODE REVIEW

**Comment by dustinmm80** [comment]
Passing in Jenkins, on a branch - https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--conjur-ami/activity?branch=publish-multiple-regions

File: ansible.sh [hunk 1]
```diff
@@ -1,8 +1,9 @@
-#!/bin/bash -ex
+#!/bin/bash -e
 
-docker build -t ansible-boto - <Dockerfile
+docker build -t ansible-boto .
```
Comment (apotterri) [inline ansible.sh]: Why do we need a build context now?

File: build.sh [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
+    ./ansible.sh ansible-playbook -v \
+      -e instance_id="$(< ${INSTANCE_ID_FILE})" \
+      -e instance_state=absent \
+      instance.yml
   fi
 }
 trap finish EXIT
 
-./ansible.sh ${@-ansible-playbook -vvv -e ami_id_filename="${AMI_ID_FILE}" -e instance_id_filename="${INSTANCE_ID_FILE}" build.yml}
+./ansible.sh ansible-playbook -vvv \
```
Comment (apotterri) [inline build.sh]: Now that this is working, we might want to dial this back to `-v`.

**Comment by dustinmm80** [comment]
AMI is now public:

screen shot 2017-09-27 at 3 50 28 pm

**Comment by dustinmm80** [comment]
Once we're happy with this we should merge to master and then run the pipeline with `PROMOTE_TO_REGIONS=true`. That will distribute the public AMIs. We have a few issues to talk about I brought up in this PR.

**Comment by dustinmm80** [comment]
Build passing, with public AMI. https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--conjur-ami/detail/publish-multiple-regions/15/pipeline

This is ready for review.

File: ansible.sh (hunk 1 above)
Comment (dustinmm80) [reply ansible.sh]: oh, i didn't recognize that's what this did. should i revert it back?

File: build.sh (hunk 2 above)
Comment (dustinmm80) [reply build.sh]: good call

File: ansible.sh (hunk 1 above)
Comment (dustinmm80) [reply ansible.sh]: nvm, i did roll it back

**Pull request merged**